lib/bolt/node/winrm.rb

```diff
@@ -333,15 +514,27 @@ def _run_task(task, input_method, arguments)
       end
 
       with_remote_file(task) do |remote_path|
-        if powershell_file?(remote_path) && stdin.nil?
-          # NOTE: cannot redirect STDIN to a .ps1 script inside of PowerShell
-          # must create new powershell.exe process like other interpreters
-          # fortunately, using PS with stdin input_method should never happen
-          output = execute("try { &""#{remote_path}"" } catch { exit 1 }")
-        else
-          path, args = *process_from_extension(remote_path)
-          output = execute_process(path, args, stdin)
-        end
+        output =
+          if powershell_file?(remote_path) && stdin.nil?
+            # NOTE: cannot redirect STDIN to a .ps1 script inside of PowerShell
+            # must create new powershell.exe process like other interpreters
+            # fortunately, using PS with stdin input_method should never happen
+            if input_method == 'powershell'
+              execute(<<-PS)
+& { # create a new scope
```

Comment:
What was the reason for using call operator `&` against the anonymous scriptblock?  I think you've covered the scope inheritance situation you were concerned about with the `$private:taskArgs` - but maybe I'm missing something?

Reply:
True, removing.